### ISSUE TITLE
[Bug] Since update to 2.1.2 - CPU temperature entity is missing

### ISSUE THREAD

**Issue by j0hnby** [tomaae/homeassistant-mikrotik_router#243]
## Describe the issue
Since updating to V 2.1.2, the CPU temperature entity isn't present for my router in homeassistant

## How to reproduce the issue
Use previous version, update to v2.1.2, it's gone.

## Expected behavior
CPU temperature measurement is present

## Screenshots
![image](https://user-images.githubusercontent.com/18377483/188474686-945bf352-7a0b-4e55-bc3b-f46251dd7651.png)

## Software versions

 - Home Assistant version: 
 - Mikrotik Router integration version: 
 - Mikrotik Hardware: 
 - RouterOS version: 

Home Assistant 2022.8.7
Supervisor 2022.08.5
Operating System 8.5
Frontend 20220802.0 - latest
Mikrotik Router integration version: v2.1.2
Mikrotik Hardware: CRS326-24G-2S+
RouterOS Version: 7.2rc4

**Comment by j0hnby** [comment]
apologies - I mistakenly pressed enter and it submitted without any details!!  Updating now

**Comment by tomaae** [comment]
I will need debugs to see what is going on there.

**Comment by j0hnby** [comment]
[home-assistant.log](https://github.com/tomaae/homeassistant-mikrotik_router/files/9491553/home-assistant.log)
Sorry - attached, thought I had done that earlier!

**Comment by tomaae** [comment]
this will be the reason:
```2022-09-05 19:00:26.437 WARNING (SyncWorker_6) [custom_components.mikrotik_router.mikrotik_controller] Mikrotik 192.168.0.1 user hass does not have sufficient access rights. Integration functionality will be limited.```
Health sensors are not available without write permissions for some reason.

See https://github.com/tomaae/homeassistant-mikrotik_router#setup-integration for both full and limited requirements.

**Comment by j0hnby** [comment]
Up until the update to 2.1.2, the api user didn't have write and showed the temperature.  This afternoon I changed it to write to test access as found that detail in the docs while doing troubleshooting.  Still no temperature measurement.  I changed it back to read then.

![image](https://user-images.githubusercontent.com/18377483/188503706-f3ec506c-911d-4528-8bc9-446b84465dc6.png)

Those logs were gathered with the read access, attached are ones with write instead
[home-assistant (with write access).log](https://github.com/tomaae/homeassistant-mikrotik_router/files/9491702/home-assistant.with.write.access.log)

**Comment by tomaae** [comment]
you are missing "policy". but it may be possible that I have forgot to adjust required permissions for system health.

**Comment by tomaae** [comment]
![image](https://user-images.githubusercontent.com/23486452/188551562-13015e08-f986-444e-9b66-424998399738.png)
this could be also related

**Comment by j0hnby** [comment]
Adding in "Policy" as the user permission brought it back - what a strange set to requirements from MKT to get env data!  
![image](https://user-images.githubusercontent.com/18377483/188580364-00106477-30ff-42d9-a37a-c2ad49756afd.png)

Thanks for your help!

**Comment by katiuskt** [comment]
Same problem here: no temperature sensor after upgrade 

Device: RB4011iGS+5HacQ2HnD
User belongs to a group with these permissions. In particular, policy and write ones are enabled:
![image](https://user-images.githubusercontent.com/75622100/189733345-732f7d80-aeb7-4ff6-b38b-8fab0c331194.png)

I can see this into HASS logs:

```
2022-09-12 20:47:34.434 DEBUG (SyncWorker_5) [custom_components.mikrotik_router.mikrotikapi] API query: /system/routerboard
2022-09-12 20:47:34.465 DEBUG (SyncWorker_5) [custom_components.mikrotik_router.apiparser] Processing source [{'routerboard': True, 'model': 'RB4011iGS+5HacQ2HnD', 'serial-number': '**REDACTED**', 'firmware-type': 'al2', 'factory-firmware': '6.45.8', 'current-firmware': '7.5', 'upgrade-firmware': '7.5'}]
2022-09-12 20:47:34.466 DEBUG (SyncWorker_5) [custom_components.mikrotik_router.apiparser] Processing entry {'routerboard': True, 'model': 'RB4011iGS+5HacQ2HnD', 'serial-number': '**REDACTED**', 'firmware-type': 'al2', 'factory-firmware': '6.45.8', 'current-firmware': '7.5', 'upgrade-firmware': '7.5'}
```

But AFAIK, temperature is received by _/system/health_ call (or at least that is the way I get that info into WinBox terminal):
```
/system/health> print
Columns: NAME, VALUE, TYPE
#  NAME         VALUE  TYPE
0  voltage      24.5   V   
1  temperature  48     C
```
And such _/system/health_ call is not printed into logs.

Temperature information was reported prior 2.1.2 upgrade (although I'm afraid I can't confirm if it was working on Mikrotik 7.4.x release or 7.5 one)

**Comment by tomaae** [comment]
no, thats why I'm leaving this open. I will need to crosscheck some permissions as I may have copied some without adjusting them.
policy should not be technically needed there as well. but I will test it all manually on several devices before changing it.
system health is really messed up on mikrotik side when it comes to persmissions.

**Comment by Scream85** [comment]
Same here. Unfortunately after adding `policy` and making sure the `reboot` also enabled in the group of the `api` user, it is still not working. Also for some reason HACS is not showing the version number of Mikrotik Router integration.
```shell
{
  "home_assistant": {
    "installation_type": "Home Assistant OS",
    "version": "2025.1.1",
    "dev": false,
    "hassio": true,
    "virtualenv": false,
    "python_version": "3.13.1",
    "docker": true,
    "arch": "aarch64",
    "timezone": "Europe/London",
    "os_name": "Linux",
    "os_version": "6.6.62-haos-raspi",
    "supervisor": "2024.12.3",
    "host_os": "Home Assistant OS 14.1",
    "docker_version": "27.2.0",
    "chassis": "embedded",
    "run_as_root": true
  },
    "mikrotik_router": {
      "documentation": "https://github.com/tomaae/homeassistant-mikrotik_router",
      "version": "0.0.0",
      "requirements": [
        "librouteros>=3.2.0",
        "mac-vendor-lookup>=0.1.12"
      ]
    },
"integration_manifest": {
    "domain": "mikrotik_router",
    "name": "Mikrotik Router",
    "config_flow": true,
    "iot_class": "local_polling",
    "documentation": "https://github.com/tomaae/homeassistant-mikrotik_router",
    "issue_tracker": "https://github.com/tomaae/homeassistant-mikrotik_router/issues",
    "dependencies": [],
    "requirements": [
      "librouteros>=3.2.0",
      "mac-vendor-lookup>=0.1.12"
    ],
    "codeowners": [
      "tomaae"
    ],
    "version": "0.0.0",
    "is_built_in": false,
    "overwrites_built_in": false
  },
        "routerboard": {
        "routerboard": true,
        "model": "RB4011iGS+5HacQ2HnD",
        "serial-number": "**REDACTED**"
      },
      "resource": {
        "platform": "MikroTik",
        "board-name": "RB4011iGS+5HacQ2HnD",
        "version": "7.13.2 (stable)",
        "uptime_str": "15w1d9h6m40s",
        "cpu-load": 2,
        "free-memory": 938774528,
        "total-memory": 1073741824,
        "free-hdd-space": 426729472,
        "total-hdd-space": 536870912,
        "uptime": "2024-09-24T05:58:44+00:00",
        "uptime_epoch": 9191200,
        "clients_wired": 0,
        "clients_wireless": 0,
        "captive_authorized": 0,
        "memory-usage": 13,
        "hdd-usage": 21
      },
      "health": {},
      "health7": {},
      "host_hass": {},
      "hostspot_host": {},
      "client_traffic": {},
      "environment": {},
      "ups": {},
      "gps": {}
    }
  }
}
```
